pkg/component/component.go

```diff
@@ -360,3 +362,91 @@ func Exists(client *occlient.Client, componentName string) (bool, error) {
 	}
 	return false, nil
 }
+
+// GetComponentService returns the Service object associated with the given
+// component.
+// An error is thrown when exactly one Service is not found for the component.
+func GetComponentService(client *occlient.Client, componentName string, applicationName string) (*corev1.Service, error) {
+	labels := GetLabels(componentName, applicationName, false)
+	selector := util.ConvertLabelsToSelector(labels)
+
+	services, err := client.GetServicesFromSelector(selector)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get Services for the selector: %v", selector)
+	}
+
+	numServices := len(services)
+	if numServices == 0 {
+		return nil, fmt.Errorf("no Service was found for the selector: %v", selector)
+	} else if numServices > 1 {
+		return nil, fmt.Errorf("multiple Services exist for the selector: %v. Only one must be present", selector)
+	}
+
+	return &services[0], nil
+}
+
+// LinkInfo contains the information about the link being created between
+// components
+type LinkInfo struct {
+	ComponentName string
+	Envs          []string
+}
+
+// Link injects connection information of the target component into the source
+// component as environment variables
+func Link(client *occlient.Client, sourceComponent, targetComponent, applicationName string) (*LinkInfo, error) {
+	// Get Service associated with the target component
+	targetService, err := GetComponentService(client, targetComponent, applicationName)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get service associated with the component %v", targetComponent)
+	}
+
+	// Generate environment variables to inject
+	linkHostKey := fmt.Sprintf("COMPONENT_%v_HOST", strings.ToUpper(targetComponent))
+	linkHostValue := targetService.Name
+
+	linkPortKey := fmt.Sprintf("COMPONENT_%v_PORT", strings.ToUpper(targetComponent))
+	linkPort, err := getPortFromService(targetService)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get port from Service %v", targetService.Name)
+	}
+	linkPortValue := fmt.Sprint(linkPort)
+
+	// Inject environment variable to source component
+	sourceDC, err := GetComponentDeploymentConfig(client, sourceComponent, applicationName)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get Deployment Config for component %v", sourceComponent)
+	}
+
+	if err := client.AddEnvironmentVariablesToDeploymentConfig(
+		[]corev1.EnvVar{
+			{
+				Name:  linkHostKey,
+				Value: linkHostValue,
+			},
+			{
+				Name:  linkPortKey,
+				Value: linkPortValue,
+			},
+		}, sourceDC); err != nil {
+		return nil, errors.Wrapf(err, "unable to add environment variables to Deployment Config %v", sourceDC.Name)
+	}
+
+	return &LinkInfo{
+		ComponentName: sourceComponent,
```

Comment:
Why are we returning just sourceComponent?

Reply:
modified LinkInfo to include both source and target components